src/gherlint/checkers/base_checker.py

```diff
@@ -1,15 +1,26 @@
 # pylint: disable=too-few-public-methods
-from typing import List
+from typing import List, Optional, Type
 
+from gherlint.options import Options
 from gherlint.reporting import Message, MessageStore, Reporter
 
 
 class BaseChecker:
     """Base class for all checkers."""
 
     MESSAGES: List[Message] = []
+    has_options = False
+    options: Optional[Options] = None
 
-    def __init__(self, reporter: Reporter):
+    def __init__(self, reporter: Reporter) -> None:
         self.reporter = reporter
         for message in self.MESSAGES:
             MessageStore().register_message(message)
+        if self.has_options:
```

Comment:
Is it necessary to have `has_options` attribute? Can we just check with `if self.options:`?

Reply:
Nice!